### ISSUE TITLE
[AVM Module Issue]: Virtual Machine settings missing

### ISSUE THREAD

**Issue by Gijsreyn** [Azure/bicep-registry-modules#1347]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/compute/virtual-machine

### Description

Dear AVM team, 

When working on AUMv2, there are two settings required when managing a Virtual Machine under it:

- AutomaticByPlatform
- bypassPlatformSafetyChecksOnUserSchedule

The last one is only missing in the windowsConfiguration variable (and Linux):

```
var windowsConfiguration = {
  provisionVMAgent: provisionVMAgent
  enableAutomaticUpdates: enableAutomaticUpdates
  patchSettings: (provisionVMAgent && (patchMode =~ 'AutomaticByPlatform' || patchMode =~ 'AutomaticByOS' || patchMode =~ 'Manual')) ? {
    patchMode: patchMode
    assessmentMode: patchAssessmentMode
  } : null
  timeZone: empty(timeZone) ? null : timeZone
  additionalUnattendContent: empty(additionalUnattendContent) ? null : additionalUnattendContent
  winRM: !empty(winRM) ? {
    listeners: winRM
  } : null
}
```

Is it possible to apply the following update as such:

```
@description('Optional. Enables customer to schedule patching without accidental upgrades.')
param bypassPlatformSafetyChecksOnUserSchedule bool = true

@description('Optional. Specifies the reboot setting for all AutomaticByPlatform patch installation operations.')
@allowed([
  'Always'
  'IfRequired'
  'Never'
  'Unknown'
])
param rebootSetting string = 'IfRequired'

var windowsConfiguration = {
  provisionVMAgent: provisionVMAgent
  enableAutomaticUpdates: enableAutomaticUpdates
  patchSettings: (provisionVMAgent && (patchMode =~ 'AutomaticByPlatform' || patchMode =~ 'AutomaticByOS' || patchMode =~ 'Manual'))
    ? {
        patchMode: patchMode
        assessmentMode: patchAssessmentMode
        automaticByPlatformSettings: {
          bypassPlatformSafetyChecksOnUserSchedule: bypassPlatformSafetyChecksOnUserSchedule
          rebootSetting: rebootSetting
        }
      }
    : null
  timeZone: empty(timeZone) ? null : timeZone
  additionalUnattendContent: empty(additionalUnattendContent) ? null : additionalUnattendContent
  winRM: !empty(winRM)
    ? {
        listeners: winRM
      }
    : null
}
```

Would love to hear from you!

**Comment by AlexanderSehr** [comment]
Hey @rahalan you're working on this, right? I guess it should be connected to your PR and the triage tag should be removed.

**Comment by Gijsreyn** [comment]
> @Gijsreyn @AlexanderSehr thanks for the issue. I will create a new PR, once the current is merged.

Looking forward to it!

**Comment by rahalan** [comment]
Will be added with PR #1515 . Thanks for the issue and the solution. @Gijsreyn in the future, feel free to create a PR on your own. No need for the AVM team to create it. Only reviewing needs to be done by the module owners or the AVM core team.

**Comment by Gijsreyn** [comment]
@rahalan , thank you very much once again. And I will! I was still under the impression during last community call, that contribution was still not available, but now I will keep it in mind and go through the contribution guide.
